src/main/java/org/terasology/logic/health/BlockDamageRenderer.java

```diff
@@ -80,8 +83,20 @@ public void renderOverlay() {
         blockSelectionRenderer.endRenderOverlay();
     }
 
-    private Integer getEffectsNumber(HealthComponent health) {
-        return Math.round((1f - (float) health.currentHealth / health.maxHealth) * 10.0f);
+
+    /**
+     * Compute the damage effect number as linear mapping from damage percentage to the range [0..10].
+     * <p>
+     * A value of 0 denotes no damage, while 10 denotes a current health of 0.
+     *
+     * @return the effect number in [0..10] linear to damage percentage
+     */
+    int getDamageEffectsNumber(HealthComponent health) {
+        Preconditions.checkArgument(health.currentHealth >= 0);
+        Preconditions.checkArgument(health.maxHealth > 0);
```

Comment:
I thought a hard crash would make sense in these cases, instead of weird effect numbers... :shrug: But we can discuss this, I'm fine with changing.